prometheus_client/multiprocess.py

```diff
@@ -25,6 +27,47 @@
 _db_pattern = re.compile(r"(\w+)_(\d+)\.db")
 
 
+class MetricsCache(object):
+    def __init__(self):
+        self.metrics = []
+        self.last_archive_duration = 0
+        self.lock = RLock()
+
+    def retrieve_metrics(self):
+        with self.lock:
+            return self.metrics
+
+    def write_metrics(self, metrics, time_elapsed=None):
+        with self.lock:
+            self.last_archive_duration = time_elapsed
+            self.metrics = metrics
+
+    def collect(self):
+        yield GaugeMetricFamily(
```

Comment:
It might be simpler to make this just return both self.metrics and this custom value

Reply:
Given that the mechanism already exists, I'd rather maintain this logical separation. We might want to add more meta-metrics like this